### PR TITLE
qt6: bump deps

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **qt/6.***

bump deps, remove unused patches and make test_package V2 compatible

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jwillikers — APPROVED** [review]

**Comment by ericLemanissier** [comment]
blocked by:
- https://github.com/conan-io/conan-center-index/issues/18162

**Review by MartinDelille — APPROVED** [review]

**Review by CJCombrink — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Comment by ericLemanissier** [comment]
@RubenRBS @jcar87 @czoido @prince-chrismc could one of you please review this PR ?

**Review by jwillikers — APPROVED** [review]

**Review by MartinDelille — APPROVED** [review]

**Review by MartinDelille — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]

**Comment by ericLemanissier** [comment]
OK, We should be good to go now. I'll refrain myself from further modifications in this PR

**Review by MartinDelille — APPROVED** [review]

**Comment by ericLemanissier** [comment]
@RubenRBS Sorry I dismissed your review earlier, but it's fully ready to be merged now. Don't hesitate to re-review !

File: recipes/qt/6.x.x/conanfile.py
Comment (dheater) [inline recipes/qt/6.x.x/conanfile.py:300]: Its my understanding that Qt6 does not support openssl 3.x yet.  There is a [blog post](https://www.qt.io/blog/moving-to-openssl-3-in-binary-builds-starting-from-qt-6.5-beta-2) announcing the intent to add support but I did not see it in the 6.5.0 or 6.5.1 release notes

**Review by dheater — APPROVED** [review]

File: recipes/qt/6.x.x/conanfile.py
Comment (ericLemanissier) [reply recipes/qt/6.x.x/conanfile.py:300]: This blog post says qt6 supports openssl 3 since qt 6.2.0. The migration mentioned in this blog post is just in the environment used to generate official qt binaries

Comment (dheater) [reply recipes/qt/6.x.x/conanfile.py:300]: Ah. I misread that. Thanks!

**Comment by ericLemanissier** [comment]
@davidsanfal @czoido could one of you please review this PR ?

**Comment by ericLemanissier** [comment]
Hey @danimtb @uilianries . Could one of you please review this PR ? it fixes https://github.com/conan-io/conan-center-index/issues/17449 (I'm not writing it in the PR description, because it would retrigger a CI build, which would delay the merge even more, and possibly stumble on an unrelated C3I failure)

**Review by uilianries — APPROVED** [review]
Conan v2 is not working, but Qt6 is not ported yet, and this PR does not focus on porting, but bumping dependencies instead.

**Pull request merged**
